### DISCUSSION TITLE
Sketch Loading : uf2conv vs picotool/rp2040tool

### DISCUSSION THREAD

**Discussion by Kyklas** [earlephilhower/arduino-pico#609] [General]
I am using Seeed XIAO RP2040 and the MBED based Arduino support has intermittent USB issue.
I was gald to find this Arduino support for RP2040.
I like the fact this support is Pico-SDK based.

I was using the Seeed and Arduino Mbed packages before hand.
They both use picotool/rp2040tool to load the program.

I would like to understand the use of uf2conv in this package.
My main issue is the 10 sec delay after the serial reset.
https://github.com/earlephilhower/arduino-pico/blob/master/tools/uf2conv.py#L340

Would it be possible to integrate/use picotool / rp2040tool ?

**Comment by earlephilhower** [comment]
Try the change in #610 .  Instead of waiting 10 seconds it now tries over and over immediately (up to 10 secs)  to mount and upload.

**Reply by Kyklas** [reply]
Will take a look, thanks for fast feedback.

**Reply by Kyklas** [reply]
Works well. Now when the device is in the bootloader state with a drive showing, it just flashes which is greatly faster.

**Comment by Kyklas** [comment]
I am not sure how arduino work but the output of the uf2conv script does not show up until it ends.
I sometime have that issue of stdout programing. Since we are waiting 10sec, it would be nice to have some progress indication.

```
uf2conv.py --serial COM5 --family RP2040 --deploy  ....
Resetting COM5
Caught exception during reset!
Converting to uf2, output size: 138240, start address: 0x2000
Flashing E: (RPI-RP2)
Wrote 138240 bytes to E:/NEW.UF2
```

One feedback I have is the `Caught exception during reset!`.
This stack behavior to reset is different from Mbed. 
On Mbed reset will occur when the host closes the serial@1200baud.
Here the reset occurs when the host opens the serial@1200baud.

It seem there will always be an error/exception during reset because of this.
Not sure how to indicate the process here.
